### ISSUE TITLE
Unexpected behavior with string as a primary key

### ISSUE THREAD

**Issue by shiasyn** [Laravel-Backpack/CRUD#1867]
# Bug report

Possible related issues:  [#179](https://github.com/Laravel-Backpack/CRUD/issues/179) [#1300](https://github.com/Laravel-Backpack/CRUD/issues/1300) 

### What I did:
1. Created model with following schema, so I want to use code column for a primary key
```
    Schema::create('page_parameters', function (Blueprint $table) {
            $table->string('code')->unique();
            $table->text('extras')->nullable();
            $table->timestamps();
        });
```
2. added primary key to my model
```
class PageParameters extends Model
{
    use CrudTrait;

    protected $table = 'page_parameters';

    protected $primaryKey = 'code';

    protected $guarded = [];
}

```
3. added following fields to my crud controller
```
        $fields = [
            [
                'name' => 'code',
                'label' => 'Code',
                'type' => 'text',
                'orderable' => false  // tried it without orderable too
            ],
            [
                'name' => 'extras',
                'label' => 'Extras',
                'type' => 'textarea'
            ]
        ];
        $this->crud->addFields($fields);

        $this->crud->addColumns($fields);

```

### What I expected to happen:
I expected to see table like this in my crud list

| code  | extras |
|-------|--------|
| index | ...    |
| about | ...    |
| ...   | ...    |

and being able to edit items via routes like this
```
example.com/admin/pageparams/{code}/edit
```

### What happened:

my elements list looks like this, although in database values is index and about as intended

| code  | extras |
|-------|--------|
| 0 | ...    |
| 0 | ...    |
| 0   | ...    |
| ....   | ...    |

when I try to click on edit button it gets me to the route ``` example.com/admin/pageparams/0/edit ``` and throws 404 error
although when I'm directly accessing page ```example.com/admin/pageparams/index/edit```
it works and I get proper crud element edit page

### What I've already tried to fix it:

I've tried to debug it and found that through the process of creating an elements list, function _data_get()_ from _vendor/laravel/framework/src/Illuminate/Support/helpers.php_ returns 0 instead of a text value 

### Backpack, Laravel, PHP, DB version:

| package  | version |
|-------|--------|
| backpack/base | 1.0.4|
| backpack/crud | 3.5.14|
| laravel/framework  |  v5.7.28  |
| database   | mariadb:10.1.38  |
| php |  7.2.16-fpm |

## Off topic:
Basically I'm trying to figure it out how create several pages for editing page parameters (with different fields on), the concept is very similar to page-manager but in my case I want to have about six pages with constant addresses (index/about/etc) and want to be able to edit them via links in a sidebar (just to put links like /admin/pageparams/{code}/edit into sidebar)
I'm still new to backpack and laravel and I have feeling that I most likely reinventing the weel with it, but i've seen a couple pagemanager usage examples in my colleagues projects and I've decided to use it for user-generated pages only in this case. I'm frustrated by all of that, so I will be grateful if you can give me a point in the right direction.

**Comment by tabacitu** [comment]
Hi @shiasyn ,

It's not a Backpack-problem per se, more of a Laravel problem.

But I'm 99% sure the reason you're getting ```0``` instead of the actual value is that you haven't defined incrementing false on your model:

```php
    public $incrementing = false;
```

It's not straightforward that that's the cause of it, I know. But when using strings as primary keys, it makes sense for you to do that. Once you do, you'll get the proper result.

Cheers!

PS. I've used PageManager like you've described too. Just add elements to the sidebar that point to editing that particular page. And make sure you always redirect back to the current entry.

**Comment by shiasyn** [comment]
@tabacitu, got it, thank you!
